arclight_installer.go

```diff
@@ -0,0 +1,197 @@
+package installer
+
+import (
+	"context"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strconv"
+	"strings"
+)
+
+type (
+	ArclightInstaller struct {
+	}
+
+	ArclightRelease struct {
+		Assets      []ArclightAssets `json:"assets"`
+		IsExpired   bool
+		PublishTime string `json:"published_at"`
+	}
+
+	ArclightAssets struct {
+		AssetsUrl   string `json:"url"`
+		AssetsName  string `json:"name"`
+		DownloadUrl string `json:"browser_download_url"`
+	}
+)
+
+var DefaultArclightInstaller = &ArclightInstaller{}
+
+var _ Installer = DefaultArclightInstaller
+
+func init() {
+	Installers["arclight"] = DefaultArclightInstaller
+}
+
+func (r *ArclightInstaller) Install(path, name string, target string) (installed string, err error) {
+	return r.InstallWithLoader(path, name, target, "")
+}
+
+func (r *ArclightInstaller) InstallWithLoader(path, name string, target string, loader string) (installed string, err error) {
+	versions, err := r.GetInstallerVersions()
+	if err != nil {
+		return "", err
+	}
+	if len(loader) == 0 {
+		var alreadyFind bool = false
+		allVersions := r.GetOnlyVersions(versions)
+		if target == "latest" {
+			loader, err = r.GetLatestVersion()
+			if err != nil {
+				return "", err
+			}
+			alreadyFind = true
+		}
+		for _, version := range allVersions {
+			if version == target {
+				loader = target
+				alreadyFind = true
+			}
+		}
+		if !alreadyFind {
+			loger.Info("not find the suitable builder, the version should be included in the following list:")
+			for i := 0; i < len(allVersions); i += 1 {
+				if versions[allVersions[i]].IsExpired {
+					loger.Info("versions:", allVersions[i], "  EXPIRED, DO NOT SUPPORT")
+				} else {
+					loger.Info("versions:", allVersions[i])
+				}
+			}
+			return "", &VersionNotFoundErr{target}
+		}
+	}
+	ExactDownloadeName := versions[loader].Assets[0].AssetsName
```

Comment:
```suggestion
	exactDownloadeName := versions[loader].Assets[0].AssetsName
```

请遵守命名规范，局部变量使用小驼峰命名法